p2p/libp2p/mutexHolder.go

```diff
@@ -0,0 +1,50 @@
+package libp2p
+
+import (
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/storage/lrucache"
+)
+
+// MutexHolder holds a cache of mutexes: pairs of (key, *sync.Mutex)
+type MutexHolder struct {
+	//generalMutex is used to assure access to the already concurrent safe lrucache is being serialized
+	generalMutex sync.Mutex
```

Comment:
is it a must to protect this ?

Reply:
yes, there is a get and eventually some put's there that must be made atomically. We do not want a go routine to set a mutex in a lru on a key that had an already set mutex there.